### DISCUSSION TITLE
Event hooks : triggers

### DISCUSSION THREAD

**Discussion by amirkhalil22** [pocketbase/pocketbase#3029] [Q&A] [answered]
Good morning,

i have articles collection and when i create new record i must select the user or users related to this article, so i want to send an email with for that or these user(s) to inform them for this article (using triggers)!

please help me. 

A++

**Comment by ganigeorgiev** [comment, marked as answer]
You can check the [Go guides](https://pocketbase.io/docs/go-overview/) or [JS guides](https://pocketbase.io/docs/js-overview/).

There are not enough details about your collections structure to be more specific, but here is a general example using the JS app hooks (_I haven't tested it and will differ depending on what fields your collection has; treat it as pseudo-code_):

```js
// pb_hooks/main.pb.js
onRecordAfterCreateRequest((e) => {
    const title = e.record.get("title")
    
    const user = $app.dao().findRecordById("users", e.record.get("author"))

    const message = new MailerMessage({
        from: {
            address: $app.settings().meta.senderAddress,
            name:    $app.settings().meta.senderName,
        },
        to: [
            {address: user.email()},
        ],
        subject: "New article created - " + title
        html:    "YOUR_HTML_BODY...",
        // bcc, cc and custom headers are also supported...
    })

    $app.newMailClient().send(message)
}, "articles")

```

**Reply by amirkhalil22** [reply]
Is that meta.senderAdress take the adress of the user who create the article

Also when i must create this code ? Hooks.server
Or +page.server ?

**Reply by ganigeorgiev** [reply]
The above code is not intended to be part of your SvelteKit app (if that's what you are using).

For more details about the app level JS hooks please read https://pocketbase.io/docs/js-overview/.

> Is that meta.senderAdress take the adress of the user who create the article

No. `$app.settings().meta.senderAddress` returns the email address configured in the _"Admin UI > Settings > Mail settings > Sender address"_.

The user address is `user.email()` (or you can access it as `user.get("email")`).

**Reply by sergio9929** [reply]
I've recently created something similar to what you need:
```js
onRecordAfterCreateRequest((e) => {
    $app.dao().expandRecord(e.record, ['responsibleId', 'operatorId'], null)

    const message = new MailerMessage({
        from: {
            address: $app.settings().meta.senderAddress,
            name: $app.settings().meta.senderName,
        },
        to: [{ address: e.record.expandedOne('responsibleId')?.email() }, { address: e.record.expandedOne('operatorId')?.email() }],
        subject: `Creado: ${e.record.get('key')}`,
        html: `<pre>${JSON.stringify(e.record, null, 2)}</pre><a href="http://localhost:3000/${e.collection.name}/${e.record.id}">Ver incidencia</a>`,
        bcc: [{ address: 'sergio.rodriguez@thelastdock.com' }]
        // bcc, cc and custom headers are also supported...
    })

    $app.newMailClient().send(message)
}, 'incidences')
```

`responsibleId` and `operatorId` are two different single relation fields. I haven't tested this with a multiple relation field, but replacing `e.record.expandedOne('responsibleId')` with `e.record.expandedAll('responsibleId')` and a loop should do the job.

Hope this points you towards the right direction :)

**Reply by amirkhalil22** [reply]
thanks i will try this and i think this will help me a lot

**Reply by amirkhalil22** [reply]
pb_hooks folder is under src or routes ???

**Reply by sergio9929** [reply]
You have to create it in the same path as pocketbase.exe, pb_data/ and pb_migrations/
Example:
```
components/
pages/
pb/
- pb_data/
- pb_hooks/
--- main.pb.js
--- mail.pb.js
- pb_migrations/
- pocketbase.exe
package.json
```

**Reply by amirkhalil22** [reply]
i put it there and there is nothing...

![image](https://github.com/pocketbase/pocketbase/assets/104743918/fc2a3b5c-e67f-4404-88da-bd57217e0950)

this is my code, laptop is my collection and "client" is the relation field coming from users collection where i want to get the email from there

can i put the adress like that ??
adress : "sportifyapp00@gmail.com"

something is wrong here ???

**Reply by sergio9929** [reply]
You should change this:
```js
console.log(record.expandedOne("client"))
```
to this:
```js
console.log(e.record.expandedOne("client"))
```

You can hardcode the address like that, but I recommend you keeping that part of the code as before and changing the email in `Admin > Settings > Mail settings`
```js
from: {
    address: $app.settings().meta.senderAddress,
    name: $app.settings().meta.senderName,
},
```

If you are running this in localhost I think you can only send emails using a SMTP server (google that) or capturing (not really sending) the emails with something like mailhog (google that too), the second one is the option I'm using.

**Reply by amirkhalil22** [reply]
Thanks a lot it works
I try it by a single relation 
Thank you brother ❤️

**Reply by amirkhalil22** [reply]
@ganigeorgiev @sergio9929 
Really thanks and big love ❤️
